collect_app/src/main/java/org/odk/collect/android/widgets/DecimalWidget.java

```diff
@@ -79,7 +79,7 @@ public DecimalWidget(Context context, FormEntryPrompt prompt, boolean readOnlyOv
         	// truncate to 15 digits max...
             String dString = nf.format(d);
             d = Double.parseDouble(dString.replace(',', '.'));
-            mAnswer.setText(d.toString());
+            mAnswer.setText(String.format("%d", d.toString()));
```

Comment:
I think you need to remove the call on toString() for it to be an improvement. I'm surprised that this works -- %d doesn't require the param to be an int? I think %f is for a double.